src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml

```diff
@@ -17,36 +17,37 @@
           = link_to(word_break(repository, 22),
             package_binaries_path(project: project, package: package, repository: repository),
             title: "Binaries for #{repository}")
-        .col-12
+        .col-12.d-flex.flex-row.flex-wrap.p-2
           - results_per_repository.each do |result|
-            .row
-              .col-4
-                %p
-                  = result.architecture
-              .col-5
-                = repository_status_icon(status: result.state)
-                %p.d-none.d-sm-inline.d-lg-none
-                  = result.state
-                - result_help = repository_info(result.state) if result.is_repository_in_db
-                - result_help ||= 'This result is outdated'
-                %p.d-none.d-lg-inline
-                  = result_help
-                %i.fa.fa-question-circle.d-lg-none.text-info{ data: { placement: 'top', toggle: 'popover', content: "#{result_help}" } }
-              .col-3
-                - if result.code.in?(['-', 'unresolvable', 'blocked', 'excluded', 'scheduled'])
-                  %i{ class: "fa #{build_status_icon(result.code)}" }
-                - else
-                  = link_to(package_live_build_log_path(project: project.to_s, package: package,
-                      repository: result.repository, arch: result.architecture), rel: 'nofollow') do
+            :ruby
+              result_help = repository_info(result.state) if result.is_repository_in_db
```

Comment:
I belive we should move this into a View Component, the logic would be pushed into the component class methods/instance variables and the view would be easier. And you get easier testing by the same price :moneybag:

Reply:
I would say the whole build results element is going to be moved to a component...
I'd like to do that as soon as I finish with the UI improvements.

Reply:
Maybe in another PR.